### PR TITLE
Ft/sse bucket creation tool

### PR DESCRIPTION
**Pull request by ploki**
convenient tool to create bucket with mandatory encryption flag

please, review the last commit, as others are part of different pull requests.

### CODE REVIEW

File: lib/kms/utilities.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    if (verbose) {
+        logger.info(` => HEADERS: ${JSON.stringify(request._headers)}`);
+    }
+    request.write('');
+    request.end();
+}
+
+
+export function createEncryptedBucket() {
+    const Getopt = require('node-getopt');
```
Comment (LaurenSpiegel) [inline lib/kms/utilities.js]: require at the top of the file

File: lib/kms/utilities.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                                verbose) {
+    const options = {
+        method: 'PUT',
+        hostname,
+        port,
+        path: `/${bucketName}/`,
+        headers: {
+            'Content-Length': 0,
+            'host': '127.0.0.1:8000',
+            'x-scal-server-side-encryption': 'AES256',
```
Comment (LaurenSpiegel) [inline lib/kms/utilities.js]: I thought we were going to do x-amz-scal-server-side-encryption

**Comment by LaurenSpiegel** [comment]
cool tool.  made a few comments.

File: package.json [hunk 3]
```diff
@@ -37,7 +37,8 @@
     "vaultclient": "scality/vaultclient",
     "werelogs": "scality/werelogs",
     "xml": "~1.0.0",
-    "xml2js": "~0.4.12"
+    "xml2js": "~0.4.12",
+    "node-getopt": "^0.2.3"
```
Comment (MathieuCassagne) [inline package.json]: There is already the module "commander" to handle CLI arguments, any particular reason to use this one ?

File: bin/create_encrypted_bucket.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
```
Comment (michael-zapata) [inline bin/create_encrypted_bucket.js]: I don't get it, is it a javascript file or a shell script?

File: lib/kms/utilities.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        headers: {
+            'Content-Length': 0,
+            'host': '127.0.0.1:8000',
+            'x-amz-scal-server-side-encryption': 'AES256',
+        },
+    };
+
+    const request = http.request(options, response => {
+        if (verbose) {
+            logger.info(` <= STATUS: ${response.statusCode}`);
```
Comment (rahulreddy) [inline lib/kms/utilities.js]: `logger.info('response status code', { statusCode: response.statusCode })`

File: lib/kms/utilities.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            logger.info(` <= HEADERS: ${JSON.stringify(response.headers)}`);
```
Comment (rahulreddy) [inline lib/kms/utilities.js]: `logger.info('response headers', { headers: response.headers });`

File: lib/kms/utilities.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        }
+        response.setEncoding('utf8');
+        response.on('data', chunk => {
+            if (verbose) {
+                logger.info(`BODY: ${chunk}`);
```
Comment (rahulreddy) [inline lib/kms/utilities.js]: Instead of logging each chunk here, log the body as whole in the end event.

File: lib/kms/utilities.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+            }
+        });
+        response.on('end', () => {
+            if (verbose) {
+                logger.info('No more data in response.');
+            }
+            if (response.statusCode >= 200 && response.statusCode < 300) {
+                logger.info('Success');
```
Comment (rahulreddy) [inline lib/kms/utilities.js]: You can add response body here.

**Comment by alexandre-merle** [comment]
Updated :)

**Comment by rahulreddy** [comment]
👍

**Comment by RachedBenMustapha** [comment]
👍

File: bin/create_encrypted_bucket.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
-#!/bin/sh
-// 2>/dev/null ; exec "$(which nodejs || which node)" "$0" "$@"
```
Comment (ploki) [inline bin/create_encrypted_bucket.js]: a simple shebang like this one doesn't work everywhere

File: bin/create_encrypted_bucket.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
...
+#!/usr/bin/env node
```
Comment (ploki) [inline bin/create_encrypted_bucket.js]: this simple shebang doesn't work everywhere

File: bin/create_encrypted_bucket.js (hunk 4 above)
Comment (alexandre-merle) [inline bin/create_encrypted_bucket.js]: It's a bash script calling itself after found the node or nodejs binary :) pretty cool :)

**Pull request merged**